stdlib/public/core/Unicode.swift

```diff
@@ -30,23 +31,6 @@ public enum UnicodeDecodingResult: Equatable {
   
   /// An indication of a decoding error.
   case error
-
-  @inlinable
-  public static func == (
-    lhs: UnicodeDecodingResult,
-    rhs: UnicodeDecodingResult
-  ) -> Bool {
-    switch (lhs, rhs) {
-    case (.scalarValue(let lhsScalar), .scalarValue(let rhsScalar)):
-      return lhsScalar == rhsScalar
-    case (.emptyInput, .emptyInput):
-      return true
-    case (.error, .error):
-      return true
-    default:
-      return false
-    }
-  }
```

Comment:
This can be synthesized by the compiler.

Reply:
This is not exactly the same, because the synthesized definition isn’t inlinable. How that will affect performance/code size of code using these APIs is difficult to predict.